### PR TITLE
Auth with functionality

### PR DESCRIPTION
**Pull request by manojsarna**
- [x] Auth Added 
- [x] Page 404 added with Lottie
- [x] Components Added

https://deploy-preview-1--smashquiz-react.netlify.app/

### Ignore Rest of the Files

image

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <>
+      <Header />
+
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="*" element={<Page404 />} />
+        <Route path="/mockman" element={<MockAPI />} />
+
+        <Route element={<PrivateRoute />}>
+          {/* <Route path="/user" element={<User />} /> */}
```
Comment (iRohitGaur) [inline src/App.jsx:28]: suggestion: remove commented code

File: src/components/header/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    <header className="sm-header">
+      <nav className="sm-nav-new">
+        <Logo />
+
+        {currentPath.pathname === currentPath.pathname ? (
+          <Search searchMob={false} />
+        ) : (
+          <div className="sm-extra-div"></div>
+        )}
+        {/* <!-- ICONS BUTTONS FOR CTA --> */}
```
Comment (iRohitGaur) [inline src/components/header/Header.jsx:21]: suggestion: remove commented code

File: src/redux/reducers/authSlice.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+export const testLogin = createAsyncThunk(
+  "auth/testLogin",
+  async (arg, { rejectWithValue }) => {
+    try {
+      const response = await axios.post("/api/auth/login", {
+        email: "manojsarna@gmail.com",
+        password: "Manoj@123",
+      });
+      console.log("check");
```
Comment (iRohitGaur) [inline src/redux/reducers/authSlice.js:50]: suggestion: remove `console.log`

File: src/redux/reducers/authSlice.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      const response = await axios.post(
+        "/api/auth/verify",
+        {},
+        {
+          headers: {
+            authorization: encodedToken,
+          },
+        }
+      );
+      console.log("verify call", response.data.user);
```
Comment (iRohitGaur) [inline src/redux/reducers/authSlice.js:72]: suggestion: remove `console.log`

File: src/redux/reducers/authSlice.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+      return response.data;
+    } catch (error) {
+      console.log("verify call error", error);
```
Comment (iRohitGaur) [inline src/redux/reducers/authSlice.js:75]: suggestion: remove `console.log`

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Gave some suggestions, please look into it.

Good to merge 👍🏻

File: src/Mockman.jsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import Mockman from "mockman-js";
+function MockAPI() {
+  document.title = "MockAPI - SmashCart - Manoj Sarna";
```
Comment (pankaj-prog) [inline src/Mockman.jsx:4]: set this in useEffect

File: src/components/category-card/CategoryCard.jsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  const [imgSrc, setImgSrc] = useState(images[0]);
+
+  let i = 0;
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if (i > images.length - 1) i = 0;
+      setImgSrc(() => images[i]);
+      i++;
+    }, 3000);
+    return () => clearInterval(interval);
```
Comment (pankaj-prog) [inline src/components/category-card/CategoryCard.jsx:15]: Good that you are cleaning interval on unmount

File: src/components/header/Header.jsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+          className={`sm-action-icon-btn-wrapper ${
+            showMobileIcons && "sm-icons-show"
+          }`}
+        >
+          <LightDarkIcon />
+
+          <AuthIcon />
+        </div>
+      </nav>
+      {currentPath.pathname === currentPath.pathname && (
```
Comment (pankaj-prog) [inline src/components/header/Header.jsx:37]: why are checking two same parameters here

File: src/pages/auth/auth.css [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+.sm-btn-text {
+  background: none;
+  border: none;
+  cursor: pointer;
+  font-size: 1.6rem;
+  padding: 1rem 1.5rem;
+  font-weight: 700;
+  color: var(--COLOR-BLACK);
+  /* transition: all ease-in 0.1s; */
```
Comment (pankaj-prog) [inline src/pages/auth/auth.css:78]: remove comment

File: src/pages/auth/components/Login.jsx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+    password: "",
+  });
+  const [errorMessage, setErrorMessage] = useState({
+    show: false,
+    message: "",
+  });
+
+  const emailRegex =
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  const passwordRegex = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*\W).{8,}$/;
```
Comment (pankaj-prog) [inline src/pages/auth/components/Login.jsx:17]: suggestion : keep them outside Login component

**Review by pankaj-prog — APPROVED** [review]

**Pull request merged**